src/pipert2/core/base/flow.py

```diff
@@ -89,10 +91,24 @@ def execute_event(self, event: Method) -> None:
 
         """
 
-        for routine in self.routines.values():
-            routine.execute_event(event)
+        event_commands = event.name.split(EVENT_SEPARATOR)
+
+        if len(event_commands) > 1:
+            can_execute = self.name == event_commands[FLOW_INDEX]
+        else:
+            can_execute = True
+
+        if can_execute:
+            execution_event = Method(name=event_commands[EVENT_INDEX], params=event.params)
+
+            if len(event_commands) == ROUTINE_CONTAINED_LENGTH:
+                routine_name = event_commands[ROUTINE_INDEX]
+                self.routines.get(routine_name).execute_event(execution_event)
+            else:
+                for routine in self.routines.values():
+                    routine.execute_event(execution_event)
```

Comment:
In general, I think that most of the code could be written in somewhere else.
I think that it should be in the flow, because the most of the code handling the event name parsing.

Maybe you could move it to the Method data class ?
IMO it should be more readable and in the right place.